cns/service/main.go

```diff
@@ -405,6 +410,25 @@ func main() {
 	configuration.SetCNSConfigDefaults(&cnsconfig)
 	logger.Printf("[Azure CNS] Read config :%+v", cnsconfig)
 
+	// We might be configured to reinitialize state from the CNI instead of the apiserver.
+	// If so, we should check that the the CNI is new enough to support the state commands,
+	// otherwise we fall back to the existing behavior.
+	if cnsconfig.InitializeFromCNI {
+		isGoodVer, err := cni.IsDumpStateVer()
+		if err != nil {
+			logger.Errorf("error checking CNI ver: %v", err)
+		}
+
+		// override the prior config flag with the result of the ver check.
+		cnsconfig.InitializeFromCNI = isGoodVer
+
+		if cnsconfig.InitializeFromCNI {
+			// Set the PodInfoVersion by initialization type, so that the
+			// PodInfo maps use the correct key schema
+			cns.GlobalPodInfoVersion = cns.InterfaceIDPodInfoVersion
```

Comment:
NIT: Can we rename it to GlobalPodInfoKeyType?   PodInfoVersion sounds more like Pod version